### PR TITLE
ci: update CI Vagrant VM IP addresses

### PR DESCRIPTION
**Pull request by nbusseneau**
VirtualBox 6.1.28 introduced new restrictions on host-only networking: https://www.virtualbox.org/wiki/Changelog-6.1#v28
Manual: https://www.virtualbox.org/manual/ch06.html#network_hostonly

> On Linux, Mac OS X and Solaris Oracle VM VirtualBox will only allow IP addresses in 192.68.56.0/21 range to be assigned to host-only adapters. For IPv6 only link-local addresses are allowed. If other ranges are desired, they can be enabled by creating /etc/vbox/networks.conf and specifying allowed ranges there. For example, to allow 10.0.0.0/8 and 192.168.0.0/16 IPv4 ranges as well as 2001::/64 range put the following lines into /etc/vbox/networks.conf:
>
>      * 10.0.0.0/8 192.168.0.0/16
>      * 2001::/64
>
> Lines starting with the hash # are ignored. Next example allows any addresses, effectively disabling range control:
>
>      * 0.0.0.0/0 ::/0

These new restrictions manifest in the form of the following issue:

```
VBoxManage: error: Code E_ACCESSDENIED (0x80070005) - Access denied (extended info not available)
VBoxManage: error: Context: "EnableStaticIPConfig(Bstr(pszIp).raw(), Bstr(pszNetmask).raw())" at line 242 of file VBoxManageHostonly.cpp
```

when running `VBoxManage hostonlyif ipconfig` command where the IP address is outside of the allowed range, as is the case in our CI:

```
10:32:07  ==> k8s1-1.21: Clearing any previously set network interfaces...
10:32:09  There was an error while executing `VBoxManage`, a CLI used by Vagrant
10:32:09  for controlling VirtualBox. The command and stderr is shown below.
10:32:09
10:32:09  Command: ["hostonlyif", "ipconfig", "vboxnet0", "--ip", "192.168.38.1", "--netmask", "255.255.255.0"]
10:32:09
10:32:09  Stderr: VBoxManage: error: Code E_ACCESSDENIED (0x80070005) - Access denied (extended info not available)
10:32:09  VBoxManage: error: Context: "EnableStaticIPConfig(Bstr(pszIp).raw(), Bstr(pszNetmask).raw())" at line 242 of file VBoxManageHostonly.cpp
```

We update the IP addresses used by Vagrant VMs to be compatible with the new restricted range so that our VMs can still be run when using VirtualBox 6.1.28 and above, without the need for the `/etc/vbox/networks.conf` workaround above.

### CODE REVIEW

**Review by pchaigno — COMMENTED** [review]
Nice find!

Why are 192.168.36.0/24 and 192.168.37.0/24 not an issue? Is that because they are `virtualbox__intnet`?

Shouldn't we also update the IP(s) used by the development VMs (`/Vagrantfile`)?

File: clustermesh-apiserver/tls.rst [hunk 1]
```diff
@@ -40,7 +40,7 @@ using an externally accessible service IP from your cluster:
 
 ::
 
-    192.168.36.11 clustermesh-apiserver.cilium.io
+    192.168.56.11 clustermesh-apiserver.cilium.io
```
Comment (pchaigno) [inline clustermesh-apiserver/tls.rst:43]: :monocle_face: I don't understand the pattern here. I expected:
- 33 -> 60
- 34 -> 61
- 35 -> 62
- 36 -> 63
- 37 -> 56 (given `/21`)

**Review by pchaigno — APPROVED** [review]

File: clustermesh-apiserver/tls.rst (hunk 1 above)
Comment (nbusseneau) [reply clustermesh-apiserver/tls.rst:43]: My pattern was:

- Try to do `3X` => `5X` if possible => this perfectly matches the CI Vagrant VMs IPs.
- Start from `60` if not possible => this perfectly matches the non-CI Vagrant VMs IPs.

But I can re-do it another way if we'd like, I mostly wanted to have a quick thing out to run tests ^^

Comment (pchaigno) [reply clustermesh-apiserver/tls.rst:43]: I don't think it matters much in the end. In any case, we'll have to be careful if we ever add a new IP range.

**Comment by nbusseneau** [comment]
Travis CI hit
```
FAIL coverage: 2.7% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/hubble/parser/threefour	0.088s
FAIL make: *** [Makefile:203: integration-tests] Error 1
```
Probably coming from the PR, I must have missed a destination IP or something.

**Comment by pchaigno** [comment]
> Probably coming from the PR, I must have missed a destination IP or something.

Fixed it, we had some hardcoded BPF events in Hubble tests, with the IP address in a binary slice.

**Review by pchaigno — APPROVED** [review]

**Comment by nbusseneau** [comment]
Thanks for the fix.

**Comment by pchaigno** [comment]
/test-1.16-netnext

**Comment by ti-mo** [comment]
This was accidentally merged, sorry for the trouble. Revert in https://github.com/cilium/cilium/pull/17898.

**Comment by nbusseneau** [comment]
Re-opened at #17900.

**Pull request merged**